donrec.php

```diff
@@ -282,11 +282,11 @@ function donrec_civicrm_mailjet_transactional_bounce($bounce_message) {
 function donrec_civicrm_tabset($tabsetName, &$tabs, $context) {
   if ($tabsetName == 'civicrm/contact/view') {
     if (CRM_Core_Permission::check('view and copy receipts') || CRM_Core_Permission::check('create and withdraw receipts')) {
-      $url = CRM_Utils_System::url( 'civicrm/donrec/tab',
-                                    "reset=1&snippet=1&force=1&cid={$context['contact_id']}" );
+      $context['contact_id'] ??= NULL;
```

Comment:
What is this crazy operator? I've never seen this before!

Reply:
The [null coalescing assignment operator](https://www.php.net/manual/en/migration74.new-features.php#migration74.new-features.core.null-coalescing-assignment-operator), available since PHP 7.4. This explicitly defines the index `contact_id` if it doesn't exist, which would otherwise cause an `Undefined index` notice.

Reply:
Ah, we can't use it in CiviCRM core because that still supports php 7.3.